### PR TITLE
add budget spending permission management

### PR DESCRIPTION
**Pull request by AnthonyBuisset**
- 🗃️ create budget projection
- ♻️ use derive_new to remove boilerplate
- ✨ create budget projection
- 🔧 track budgets table in hasura
- 🗃️ make request_id mandatory in payments projection
- ✨ implement find_by_id in diesel repository
- ✨ event source budget and update projection on spend
- 🗃️ link payment request projection to budget instead of project
- 🔥 remove project exists spec
- 🔥 remove access control check when requesting payment
- ✨ spend budget when requesting payment
- 🎨 run cargo fmt after rebase
- 🔧 fix hasura metadata inconsistencies
- 🚨 ignore clippy warning
- 🗃️ create budget_spenders table
- 🔧 map budget_spenders table to hasura auth
- ♻️ split Entity trait from Aggregate
- ✨ new macro to implement mapping repository
- ♻️ use macro to implement project lead repository
- ✨ create budget spender repository
- ✨ create Budget::SpenderAssigned event and project it
- ✨ decode budgets owned permission from headers
- 🔒️ make sure the user is allowed to spend budget
- ♻️ ✨ merge create project and assign leader usecase + assign budget spender

### CODE REVIEW

File: testing/cypress/e2e/retool.cy.js [hunk 1]
```diff
@@ -17,28 +17,28 @@ describe("As an admin, on retool, I", () => {
...
-                        .its('body.data.projects_by_pk.project_leads')
-                        .its(0)
-                        .its('user_id')
-                        .should('equal', $user.id);
+                    .its('body.data.projects_by_pk.project_leads')
+                    .its(0)
+                    .its('user_id')
+                    .should('equal', $user.id);
+
+                cy.request("POST", "/v1/graphql", {
```
Comment (ofux) [inline testing/cypress/e2e/retool.cy.js:33]: why not using `cy.graphql(` here?

**Review by ofux — APPROVED** [review]

File: domain/src/user/id.rs [hunk 2]
```diff
@@ -15,5 +22,17 @@ use serde::{Deserialize, Serialize};
...
+
+impl<DB> ToSql<diesel::sql_types::Uuid, DB> for Id
+where
+	DB: Backend,
+	Uuid: ToSql<diesel::sql_types::Uuid, DB>,
+{
+	fn to_sql<W: Write>(&self, out: &mut Output<W, DB>) -> serialize::Result {
+		self.0.to_sql(out)
+	}
+}
```
[the remark this answers is not in the document]
Comment (AnthonyBuisset) [reply domain/src/user/id.rs:38]: done

File: domain/src/budget/id.rs [hunk 3]
```diff
@@ -1,8 +1,38 @@
...
+
+impl<DB> ToSql<diesel::sql_types::Uuid, DB> for Id
+where
+	DB: Backend,
+	Uuid: ToSql<diesel::sql_types::Uuid, DB>,
+{
+	fn to_sql<W: Write>(&self, out: &mut Output<W, DB>) -> serialize::Result {
+		self.0.to_sql(out)
+	}
+}
```
[the remark this answers is not in the document]
Comment (AnthonyBuisset) [reply domain/src/budget/id.rs]: done

File: testing/cypress/e2e/retool.cy.js (hunk 1 above)
Comment (AnthonyBuisset) [reply testing/cypress/e2e/retool.cy.js]: done

**Pull request merged**